src/compiler/binder.ts

```diff
@@ -2768,8 +2768,8 @@ namespace ts {
 
         function bindExportAssignment(node: ExportAssignment) {
             if (!container.symbol || !container.symbol.exports) {
-                // Export assignment in some sort of block construct
-                bindAnonymousDeclaration(node, SymbolFlags.Alias, getDeclarationName(node)!);

```

Comment:
in hindsight, creating an alias for an incorrect declaration is *extremely* likely to fail later.